src/utils/search/search.service.ts

```diff
@@ -110,10 +110,17 @@ export class SearchService {
     let results: ISearchResultEntry[] = [];
     results = await this.buildSearchResults(userResults);
     results.push(...(await this.buildSearchResults(groupResults)));
-
+    this.ensureUniqueTermsPerResult(results);
     return results;
   }
 
+  ensureUniqueTermsPerResult(results: ISearchResultEntry[]) {
```

Comment:
This might not work if there are many results. array.forEach is parallel in JS and if you assign a value to a variable this might be a problem.